### PR TITLE
Replace lodash utils

### PR DESCRIPTION
**Pull request by frankychung**
From #112 -- Removing lodash utils decreases the build size significantly as described by @dariocravero so this PR replaces the remaining three lodash modules with basic utils.

### CODE REVIEW

File: src/utils/identity.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export default function identity(value) {
+  return value;
+}
```
Comment (chicoxyzzy) [inline src/utils/identity.js]: actually lodash's identity function is the same as yours

Comment (goatslacker) [inline src/utils/identity.js]: It's probably best to put these all in a single file and export them individually that way you avoid the overhead of including each file, I think it adds ~3 lines and a bunch of bytes for every file you include.

This also has the benefit of keeping all the utils together.

Comment (goatslacker) [inline src/utils/identity.js]: As an aside: identity would then be something like:

`export const identity = x => x;` which is just :sunglasses:

**Comment by taylorhakes** [comment]
These functions are only used in one spot each. I think it would be better to keep the code inline and not create an unnecessary abstraction in another file. If they start to get used a lot or become a source of bugs, then put them in another file.

File: src/utils/identity.js (hunk 1 above)
Comment (frankychung) [inline src/utils/identity.js]: @goatslacker Grouping these in one file sounds like a good idea to me however I worry it doesn't follow the style of the rest of utils of exporting one function per module (or as @taylorhakes mentions inlining some of them, especially simple ones like identity). Also, exporting arrow functions is pleasantly succinct but also another question of style :). @gaearon thoughts?

File: src/utils/isPlainObject.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function isPlainObject(obj) {
+  return typeof obj === 'object' && Object.getPrototypeOf(obj) === Object.prototype;
```
Comment (ooflorent) [inline src/utils/isPlainObject.js]: This one does not work.

```js
isPlainObject(null) // TypeError: Object.getPrototypeOf called on non-object
```

File: src/utils/identity.js (hunk 1 above)
Comment (ooflorent) [inline src/utils/identity.js]: As @goatslacker suggested, using an arrow function would be better because:

- `identity` is not `new`-able
- `identity` has no `prototype`

File: src/utils/isPlainObject.js (hunk 2 above)
Comment (frankychung) [inline src/utils/isPlainObject.js]: Thank you, added another test case for calling with `null`

File: src/utils/isPlainObject.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default function isPlainObject(obj) {
+  return obj ? typeof obj === 'object' && Object.getPrototypeOf(obj) === Object.prototype : false;
```
Comment (goatslacker) [inline src/utils/isPlainObject.js]: A simple way to detect a pojo is to do Object.prototype.toString.call(object) === '[object Object]' or something like that

Comment (ooflorent) [inline src/utils/isPlainObject.js]: I won't work:

```js
var F = (function(){})
var f = new F
Object.prototype.toString.call(f) // [object Object]
```

File: src/utils/identity.js (hunk 1 above)
Comment (emmenko) [inline src/utils/identity.js]: > It's probably best to put these all in a single file and export them individually

:+1: 

> using an arrow function would be better

I think it's just a matter whether we want to have it as a named function or not.

Comment (ooflorent) [inline src/utils/identity.js]: > I think it's just a matter whether we want to have it as a named function or not.

It is not the only implication of arrow functions ;)

File: src/utils/isPlainObject.js (hunk 3 above)
Comment (taylorhakes) [inline src/utils/isPlainObject.js]: @ooflorent Are we trying to avoid the object above or `Object.create(null)`? It is perfectly fine for someone to have a React state that is not a plain object. ImmutableJS is used for state and it is not a plain object.

Comment (gaearon) [inline src/utils/isPlainObject.js]: >It is perfectly fine for someone to have a React state that is not a plain object.

I thought that broke in some recent version, didn't it? Either way they don't officially support it yet.
https://github.com/facebook/react/issues/3303

Comment (gaearon) [inline src/utils/isPlainObject.js]: What I want from `isPlainObject`:

* no to arrays
* no to null or undefined
* no to ImmutableJS records

I don't really care for edge cases like `Object.create(null)`.

Comment (taylorhakes) [inline src/utils/isPlainObject.js]: @gaearon I retract my earlier comment. You are correct

**Comment by gaearon** [comment]
This looks good to me. I want to go ahead with #112 so merging.
Thanks!
